### PR TITLE
enhancement(remap): Add to_syslog_level function for parsing Syslog levels into strings

### PR DESCRIPTION
**Pull request by lucperkins**
Addresses #5500 with a Remap function that converts Syslog severity levels into their corresponding keyword.

**Note**: the heavy lifting has been done here. I'm marking this as a draft until (a) #5522 lands and (b) we figure out a naming system for modules/namespaces.

### CODE REVIEW

**Comment by jszwedko** [comment]
Same as my comment in #5501 . I propose we at least call this `to_syslog_level` but #5507 would have it be `syslog::to_level`.

**Comment by lucperkins** [comment]
@jszwedko Personally, I'm agnostic on this. Happy to go with the flow re: decisions from our fearless TRL leaders.

**Review by jszwedko — COMMENTED** [review]
In absence of https://github.com/timberio/vector/issues/5507, which feels like it might be a little ways off, I'd propose calling this `to_syslog_level`. We can always deprecate it and change to `syslog::to_level` later.

Otherwise, this looks good to me!

File: lib/remap-functions/Cargo.toml [hunk 1]
```diff
@@ -135,7 +136,9 @@ strip_whitespace = []
 to_bool = ["shared/conversion"]
 to_float = ["shared/conversion"]
 to_int = ["shared/conversion"]
+to_level = []
```
Comment (jszwedko) [inline lib/remap-functions/Cargo.toml:139]:
```suggestion

```

File: lib/remap-functions/src/to_syslog_level.rs [hunk 2]
```diff
@@ -0,0 +1,151 @@
+use remap::prelude::*;
+
+#[derive(Clone, Copy, Debug)]
+pub struct ToSyslogLevel;
+
+impl Function for ToSyslogLevel {
+    fn identifier(&self) -> &'static str {
+        "to_level"
```
Comment (jszwedko) [inline lib/remap-functions/src/to_syslog_level.rs:8]:
```suggestion
        "to_syslog_level"
```

File: docs/reference/remap/to_syslog_level.cue [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+		Converts a Syslog [severity level](\(urls.syslog_levels)) into its corresponding keyword,
+		i.e. 0 into `"emerg"`, 1 into `"alert", etc.
+		"""
+	examples: [
+		{
+			title: "Success"
+			input: {
+				severity: "5"
+			}
+			source: ".log_level = to_level(.severity)"
```
Comment (jszwedko) [inline docs/reference/remap/to_syslog_level.cue:24]:
```suggestion
			source: ".log_level = to_syslog_level(.severity)"
```

This is making me realize it'd be cool to have these examples run in CI :smile:

File: docs/reference/remap/to_syslog_level.cue [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+			output: {
+				level: "notice"
+			}
+		},
+		{
+			title: "Error"
+			input: {
+				severity: "1337"
+			}
+			source: ".log_level = to_severity(.severity)"
```
Comment (jszwedko) [inline docs/reference/remap/to_syslog_level.cue:34]:
```suggestion
			source: ".log_level = to_syslog_level(.severity)"
```

**Review by jszwedko — APPROVED** [review]
We need to update the function name in a few more places, but otherwise this looks good!

**Pull request merged**